### PR TITLE
Fix to check standard_name values on quality control variables only f…

### PR DESCRIPTION
**Pull request by ggalibert**
…or variables that actually uses a standard_name.

There's no need to check qc variable standard names values for variables that don't have standard names!

### CODE REVIEW

**Comment by ggalibert** [comment]
Don't merge yet, will work on unit tests for this.

**Comment by ggalibert** [comment]
@mhidas Ready to be reviewed. Unit test and netcdf file fixed.

Modification on the netcdf file are as follow:

```
ggalibert@10-nsp-mel:~/compliance-checker$ diff /tmp/imos_variable_test.dump /tmp/imos_variable_test.dumpfixed
18a19
>                 LONGITUDE:ancillary_variables = "LONGITUDE_quality_control" ;
```

File: compliance_checker/imos/imos.py [hunk 1]
```diff
@@ -1395,16 +1395,20 @@ def check_quality_control_variable_standard_name(self, ds):
...
                     result_name = ('var', 'quality_variable', qc_variable.name,\
-                                    data_variable.name, 'check_standard_name')
-                    if getattr(qc_variable, 'standard_name', '') != value:
-                        reasoning = ["Standard name is not correct"]
-                        result = Result(BaseCheck.HIGH, False, result_name, reasoning)
+                                    variable.name, 'check_standard_name')
+                    if value:
+                        value = value + ' ' + 'status_flag'
+                        if getattr(qc_variable, 'standard_name', '') != value:
+                            reasoning = ["Standard name is not correct"]
```
Comment (mhidas) [inline compliance_checker/imos/imos.py]: How about `reasoning = ["standard_name should be '%s'." % value]` ?

**Comment by ggalibert** [comment]
Thank, took your last comment into account.

**Comment by mhidas** [comment]
:+1: Thanks.

**Pull request merged**